tabled/src/settings/split/mod.rs

```diff
@@ -0,0 +1,174 @@
+//! Split setting is a work in progress
+
+use crate::records::{ExactRecords, Records, Resizable};
+
+use super::TableOption;
+
+#[derive(Debug, Clone, Copy)]
+enum Direction {
+    Column,
+    Row,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum Behavior {
+    Append,
+    Zip,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct Split {
+    direction: Direction,
+    behavior: Behavior,
+    index: usize,
+}
+
+impl Split {
+    pub fn column(index: usize) -> Self {
+        Split {
+            direction: Direction::Column,
+            behavior: Behavior::Zip,
+            index,
+        }
+    }
+
+    pub fn row(index: usize) -> Self {
+        Split {
+            direction: Direction::Row,
+            behavior: Behavior::Zip,
+            index,
+        }
+    }
+
+    pub fn set_behavior(self, behavior: Behavior) -> Self {
```

Comment:
Just a thought.

What do you think about not exposing the `Behavior` and create a method `append`/`zip`.

```
    pub fn append(self) -> Self {
        Self { behavior: Behavior::Append, ..self }
    }

   pub fn zip(self) -> Self {
        Self { behavior: Behavior::Zip,  ..self }
    }
```

`zip` method kind of sounds good but `append` might be misleading though....

Reply:
Great suggestion 😄 

 I'm not sure what to rename append to though. Perhaps `join` or `concatenate`? I want it to be predictable.

Reply:
https://github.com/zhiburt/tabled/pull/299/files#diff-881689aca954c8702c2bde0c3bf79fb1240972e785c9e672358dcc1407c05e33R43-R55

Reply:
concatenate is good?
(shorten to concat?)

Reply:
I like it!